File: Retos/Reto #0 - EL FAMOSO FIZZ BUZZ [Fácil]/lua/bighelmet7.lua
--- Fizzbuzz
---@param num integer that represent the max range to execute
local function fizzbuzz(num)
    for i = 1, num do
        if i % 3 == 0 and i % 5 == 0 then
            print("fizzbuzz")
        elseif i % 3 == 0 then
            print("fizz")
        elseif i % 5 == 0 then
            print("buzz")
        else
            print(i)
        end
    end
end

fizzbuzz(100)

File: Retos/Reto #0 - EL FAMOSO FIZZ BUZZ [Fácil]/rust/bighelmet7.rs
fn main() {
    let num = 100;
    fizzbuzz(num);
}

fn fizzbuzz(num: i32) {
    if num == 0 {
        return;
    }

    fizzbuzz(num - 1);

    match (num % 3 == 0, num % 5 == 0) {
        (true, true) => println!("fizzbuzz"),
        (true, false) => println!("fizz"),
        (false, true) => println!("buzz"),
        _ => println!("{num}"),
    }
}
